### PR TITLE
Make kubelet to block on network plugin initialization

### PR DESCRIPTION
**Pull request by pravisankar**

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/networking PTAL

File: plugins/osdn/common.go [hunk 1]
```diff
@@ -214,10 +214,27 @@ func (oc *OvsController) StartNode(mtu uint) error {
 		return fmt.Errorf("Failed to start plugin: %v", err)
 	}
 
-	if oc.ready != nil {
-		close(oc.ready)
+	oc.MarkPodNetworkReady()
```
Comment (dcbw) [inline plugins/osdn/common.go]: MarkPodNetworkReady() doesn't need to be accessed from outside the OVSController, right?  Maybe we should lower-case the 'M' then so it's not exported.

File: plugins/osdn/common.go [hunk 2]
```diff
@@ -214,10 +214,27 @@ func (oc *OvsController) StartNode(mtu uint) error {
...
+
+	return nil
+}
+
+func (oc *OvsController) MarkPodNetworkReady() {
+	if oc.podNetworkReady != nil {
```
Comment (dcbw) [inline plugins/osdn/common.go]: It looks like we're always setting up oc.podNetworkReady, so we don't need to check for nil at all here.

**Comment by pravisankar** [comment]
Updated

**Comment by danwinship** [comment]
So this lets us get rid of the k8s patch, which is good, but it still requires the goroutine goofiness discussed in #209. It would be nice if we could untangle that somehow. (Maybe later.)

**Comment by danwinship** [comment]
lgtm

**Comment by dcbw** [comment]
LGTM

**Pull request merged**